src/plugins/analysis/file_system_metadata/code/file_system_metadata.py

```diff
@@ -129,16 +137,48 @@ def __init__(self):
     def analyze(self, file_handle: FileIO, virtual_file_path: dict, analyses: dict[str, BaseModel]) -> Schema:
         del virtual_file_path
         file_type = analyses['file_type'].mime
-        result = self._extract_metadata(file_handle, file_type)
+        result = self._extract_metadata(file_handle, file_type, analyses)
         return self.Schema(files=result)
 
-    def _extract_metadata(self, file_handle: FileIO, file_type: str) -> list[FileMetadata]:
+    def _extract_metadata(self, file_handle: FileIO, file_type: str, analyses) -> list[FileMetadata]:
         if file_type in FS_MIME_TYPES:
             return self._extract_metadata_from_file_system(file_handle)
         if file_type in ARCHIVE_MIME_TYPES:
             return _extract_metadata_from_tar(file_handle)
+        if file_type == 'filesystem/yaffs':
+            return self._extract_metadata_from_yaffs(analyses)
         return []
 
+    def _extract_metadata_from_yaffs(self, analyses) -> list[FileMetadata]:
+        result = []
+        unpacker_result = analyses.get('unpacker')
+        if not unpacker_result or unpacker_result['plugin_used'] != 'YAFFS':
+            return result
+        """
+        the output of unyaffs from the unpacker log has the following structure:
+        brw-r-----  31,   4 2014-07-19 01:28 dev/mtdblock4
+        -rw-r--r--       38 2014-07-19 01:31 build.prop
+        drwxr-xr-x        0 2016-10-11 03:12 sbin
+        lrwxrwxrwx        0 2014-07-19 01:28 sbin/ifconfig -> ../bin/busybox
+        we ignore directories and symlinks
```

Comment:
I'm not entirely conviced that it is a wise idea to ignore symlinks. It is true that most unix systems ignore symlink permissions, since they are so transparent that ownership and read changes affect the linked file instead.

_Yet_, which permissions are checked when we _create_ or _overwrite_ a symlink? Those permissions are inherited from the parent directory. Meaning that once users have write access to the parent directory, they can overwrite the symlink. This is, indeed, a security-relevant path and it is a pity that this information gets lost here.

Reply:
Since directories have no real representation in FACT (analogous to files which are represented as FileObjects), I don't really see a way to fix this. 

We could simply add them to the result of the archive/FS (where they are included), though. We could even check the permissions for inconsistencies. IMHO that would be out of scope for this PR, though.